app/views/QRScan.js

```diff
@@ -0,0 +1,291 @@
+import React, { Component } from 'react';
+import {
+  BackHandler,
+  Dimensions,
+  ImageBackground,
+  StatusBar,
+  StyleSheet,
+  View,
+} from 'react-native';
+import { RNCamera } from 'react-native-camera';
+import QRCodeScanner from 'react-native-qrcode-scanner';
+import { SvgXml } from 'react-native-svg';
+
+import BackgroundImage from './../assets/images/launchScreenBackground.png';
+import StateAtRisk from './../assets/svgs/stateAtRisk';
+import StateNoContact from './../assets/svgs/stateNoContact';
+import { isPlatformAndroid } from './../Util';
+import { Button } from '../components/Button';
+import { Typography } from '../components/Typography';
+import Colors from '../constants/colors';
+import fontFamily from '../constants/fonts';
+import languages from '../locales/languages'; // TODO internationalization
+import LocationServices from '../services/LocationService';
+
+const StateEnum = {
+  DEFAULT: 0,
+  SCAN_SUCCESS: 1,
+  SCAN_FAIL: 2,
+  SCAN_IN_PROGRESS: 3,
+};
+
+const StateIcon = ({ status, size }) => {
+  let icon;
+  switch (status) {
+    case StateEnum.DEFAULT:
+      icon = null;
+      break;
+    case StateEnum.SCAN_SUCCESS:
+      icon = StateNoContact;
+      break;
+    case StateEnum.SCAN_FAIL:
+      icon = StateAtRisk;
+      break;
+    case StateEnum.SCAN_IN_PROGRESS:
+      icon = StateNoContact;
+      break;
+  }
+  return (
+    <SvgXml xml={icon} width={size ? size : 80} height={size ? size : 80} />
+  );
+};
+
+const height = Dimensions.get('window').height;
+
+class QRScan extends Component {
```

Comment:
Could we convert this to a function component with hooks? The app is kind of split between class and function components at the moment, but I think the long term goal is to move towards function components.